library/core/src/ffi/c_str.rs

```diff
@@ -169,24 +182,45 @@ impl fmt::Debug for CStr {
     }
 }
 
-#[stable(feature = "cstr_default", since = "1.10.0")]
-impl Default for &CStr {
-    fn default() -> Self {
-        const SLICE: &[c_char] = &[0];
-        // SAFETY: `SLICE` is indeed pointing to a valid nul-terminated string.
-        unsafe { CStr::from_ptr(SLICE.as_ptr()) }
+/// Converts a string literal to a `&'static Cstr`.
+///
+/// # Panics
```

Comment:
This should really be a compile-time error.

Reply:
One probably simple way to achieve this is to just set the value generated by this to a constant:

```
{
    const CONST_CHECK: &'static CStr = /* ... */;
    CONST_CHECK
}
```

You could also use the unstable const blocks.